File: apple/Sources/ReactNativeWatchOS/Modifiers/TextScale.swift
import SwiftUI

/// SwiftUI `.textScale(_:)`. Scales text relative to the inherited font —
/// `'default'` (standard) or `'secondary'` (relatively smaller).
///
/// `Text.Scale` and the `textScale(_:)` modifier are watchOS 11.0+, so
/// this is gated with `#available`. On earlier versions the view is
/// returned unchanged (no-op).
enum RNWTextScaleModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("textScale") { view, params, _ in
            guard #available(watchOS 11.0, *) else { return view }
            let scale: Text.Scale =
                params.string("value") == "secondary" ? .secondary : .default
            return AnyView(view.textScale(scale))
        }
    }
}

File: src/modifiers/textScale.ts
import { createModifier } from './createModifier';

/// SwiftUI `Text.Scale` values. `'default'` is the standard text size;
/// `'secondary'` renders at a relatively smaller scale (e.g. for
/// superscript-like emphasis).
export type TextScale = 'default' | 'secondary';

/// SwiftUI `.textScale(_:)`. Scales text relative to the inherited font.
///
/// Requires watchOS 11.0+. On earlier versions the native side leaves the
/// view unchanged (no-op).
export function textScale(value: TextScale = 'default') {
  return createModifier('textScale', { value });
}
